### PR TITLE
Migrate to a newer JDK

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR migrates jdk version from 11 to 21 🤞🏽 .

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: engine/schema/pom.xml [hunk 1]
```diff
@@ -61,9 +61,9 @@
     <build>
         <plugins>
             <plugin>
-                <groupId>org.codehaus.gmaven</groupId>
-                <artifactId>gmaven-plugin</artifactId>
-                <version>1.5</version>
+                <groupId>org.codehaus.gmavenplus</groupId>
+                <artifactId>gmavenplus-plugin</artifactId>
+                <version>3.0.2</version>
```
Comment (DaanHoogland) [inline engine/schema/pom.xml:66]: can this come from the parent pom?

File: pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
         <sonar.host.url>https://sonarcloud.io</sonar.host.url>
 
         <!-- Build properties -->
-        <cs.jdk.version>11</cs.jdk.version>
+        <cs.jdk.version>17</cs.jdk.version>
```
Comment (DaanHoogland) [inline pom.xml:58]: if we do 21 we have more time to think of our sins.
```suggestion
        <cs.jdk.version>21</cs.jdk.version>
```

**Review by DaanHoogland — COMMENTED** [review]
good start let's see this through (cc @JoaoJandre )

File: pom.xml (hunk 2 above)
Comment (DaanHoogland) [reply pom.xml]: so this remains to be seen. depending on availabilities of libraries

**Comment by rohityadavcloud** [comment]
@vishesh92 @weizhouapache can this be closed if https://github.com/apache/cloudstack/pull/8411 is melded with https://github.com/apache/cloudstack/pull/8497 ?

**Comment by vishesh92** [comment]
> @vishesh92 @weizhouapache can this be closed if #8411 is melded with #8497 ?

This PR was for migrating the JDK and upgrade the packages required for that as well. This will require a lot of work. 

I am closing this PR for now. I will try to do this later in a way which includes small changes instead of a big PR. Most likely for 4.21.

**Pull request closed without merging**